pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1262,6 +1262,21 @@ func (c *Cluster) startMon(m *monConfig, schedule *MonScheduleInfo) error {
 		logger.Debugf("adding host path volume source to mon deployment %s", d.Name)
 	}
 
+	// Expand mon PVC if storage request for mon has increased in cephcluster crd
+	if pvcExists && c.monVolumeClaimTemplate(m) != nil {
+		desiredPvc, err := c.makeDeploymentPVC(m, false)
+		if err != nil {
+			return errors.Wrapf(err, "failed to make mon %q pvc", d.Name)
+		}
+
+		existingPvc, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.Namespace).Get(ctx, m.ResourceName, metav1.GetOptions{})
+		if err == nil {
```

Comment:
How about this structure instead?

```go
if err != nil && !kerrors.IsNotFound(err){
    return errors.Wrapf(err, "failed to fetch pvc for mon %q", m.ResourceName)
}
controller.ExpandPVCIfRequired(c.context.Client, desiredPvc, existingPvc)
```

Reply:
what if not found

Reply:
Then just return on `if err != nil`? Why would we need to check for `IsNotFound` at all if we don't handle it differently than  `err != nil`?

Reply:
Done